app/uploaders/simple_forms_api/form_remediation/uploader.rb

```diff
@@ -31,6 +31,18 @@ def store_dir
         @directory
       end
 
+      def store!(file)
+        if file.nil? || !file.respond_to?(:filename)
+          config.handle_error('Invalid file object provided for upload. Skipping.')
+          return
+        end
+
+        super(file)
+      rescue Aws::S3::Errors::ServiceError => e
+        config.handle_error("Upload failed for #{file.filename}. Enqueuing for retry.", e)
```

Comment:
Doesn't `handle_error` raise the error? Would that cease execution such that the `UploadRetryJob.perform_async` is never reached? Can we add a test to demonstrate `UploadRetryJob.perform_async` being called?

Reply:
Oops, excellent point. Fixed! 👍

Reply:
> Can we add a test to demonstrate UploadRetryJob.perform_async being called?

Good call, I'll add that!

Reply:
I went ahead and added comprehensive test coverage, including for the `UploadRetryJob.perform_async` call.